### DISCUSSION TITLE
`helmfile template` to support a `--show-only` parameter to pass to `helm template`

### DISCUSSION THREAD

**Discussion by jim-barber-he** [helmfile/helmfile#1493] [Ideas]
In the `Towards Helmfile 1.0` document ( https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md ) item 3 is to remove the `--args` parameter from helmfile stating that use-cases should be covered.
I have a use-case that looks like it isn't.

We have a helm chart for a PHP application that has a number of configuration files that are rendered into a Kubernetes Secret by the chart.
Our pipeline renders some of these configuration files using appropriate environment variables and then runs a lint against them.
This checks for things like if one of the environment variables needed is missing or malformed as well as any other malformed content that could be in the configuration file.
The rendered configuration files are also stored as build artifacts and used in later build steps when running unit tests.

The relevant part of the shell script in the pipeline that renders these configuration files does this:
```bash
# Get the template files from the Helm chart that creates the config-files secret.
# This also renders the environment variables within them.
CONFIG_FILES=$(helmfile -l name=heweb template --skip-deps --args '--show-only templates/config-files.yaml')

# Extract the templates we want.
echo "$CONFIG_FILES" | awk '/^  heweb-constants\.php:/ {print $NF}' | sed -r 's/^"|"$//g' | base64 -d > g/constants.php
echo "$CONFIG_FILES" | awk '/^  heweb-database\.php:/ {print $NF}' | sed -r 's/^"|"$//g' | base64 -d > g/database.php
echo "$CONFIG_FILES" | awk '/^  heweb-server\.php:/ {print $NF}' | sed -r 's/^"|"$//g' | base64 -d > g/server.php
echo "$CONFIG_FILES" | awk '/^  heweb-env:/ {print $NF}' | sed -r 's/^"|"$//g' | base64 -d > .env
```

It would be good to be able to replace:
```
helmfile -l name=heweb template --skip-deps --args '--show-only templates/config-files.yaml'
```
with:
```
helmfile -l name=heweb template --skip-deps --show-only templates/config-files.yaml
```
And have helmfile pass the `--show-only templates/config-files.yaml` part to the `helm template` command that it runs.

**Comment by jim-barber-he** [comment]
I've attempted a pull request to implement the above here: https://github.com/helmfile/helmfile/pull/1494

**Comment by jouve** [comment]
didn't know about --show-only, so I use yq extract/work with the data yaml:
```shell
CONFIG_FILES=$(
    helmfile -l name=heweb template --skip-deps | 
    yq 'select(.kind == "Secret" and .metadata.name == "...") | .data'
)
yq '.["heweb-constants.php"]|@base64d' <<<"$CONFIG_FILES" > g/constants.php
```

**Reply by jim-barber-he** [reply]
Thanks for the tip (in case I can't get my pull request over the line).
Which `yq` is in use above out of interest? https://github.com/mikefarah/yq or https://github.com/kislyuk/yq ?
I believe they support different syntax.

**Reply by jim-barber-he** [reply]
Well it certainly works with https://github.com/mikefarah/yq (it's the one I have available to me by default, so that's good for me)
Thanks.

**Reply by jouve** [reply]
yes, I use https://github.com/mikefarah/yq :)

**Comment by jim-barber-he** [comment]
My pull request has been approved and merged. :raised_hands:
